app/assets/javascripts/govuk_publishing_components/analytics-ga4/gtm-click-tracking.js

```diff
@@ -25,11 +25,25 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
           link_url: window.location.href.substring(window.location.origin.length),
           ui: JSON.parse(target.getAttribute('data-gtm-attributes')) || {}
         }
-        var ariaExpanded = this.checkExpandedState(target)
-        if (ariaExpanded) {
-          data.ui.state = ariaExpanded === 'false' ? 'opened' : 'closed'
+        var ariaExpanded = this.getClosestAttribute(target, 'aria-expanded')
+        /*
+          the details component uses an 'open' attribute instead of aria-expanded, so we need to check if we're on a details component.
+          since details deletes the 'open' attribute when closed, we need this boolean, otherwise every element which
+          doesn't contain an 'open' attr would be pushed to gtm as a closed element.
+        */
+        var canHaveOpenAttribute = target.getAttribute('data-module') ? target.getAttribute('data-module').includes('govuk-details') : false
```

Comment:
I think it should work if you do the following (tested and it seems to work)

```
var detailsElement = target.closest('details')
```

It's a little inefficient, because `closest` will traverse the whole way up the DOM until it finds what its looking for or runs out of elements. But I think if we tried to be more specific (like assuming the details element is one level up) we'd make the code too brittle, so I think this works.

Reply:
Thanks @andysellick - i've updated this, and I think I've rewritten the test so it makes sense.